File: julia/2017/6/solution1.jl
f = open("../../../data/2017/6/data.txt")
banks = map(x -> parse(Int, x), split(readstring(f), "\t"))
close(f)

configs = []
cycles = 0
nbanks = length(banks)
while true
    cycles += 1
    ind = indmax(banks)
    blocks = banks[ind]
    banks[ind] = 0
    for _ = blocks:-1:1
        ind += 1
        if ind > nbanks
            ind = 1
        end
        banks[ind] += 1
    end
    if banks in configs
        println("Solution is $cycles")
        break
    else
        push!(configs, copy(banks))
    end
end

File: python/2017/6/solution1.py
with open("../../../data/2017/6/data.txt", "rb") as f:
    banks = [int(x.strip()) for x in f.read().split()]

configs = []
cycles = 0
nbanks = len(banks)
while True:
    cycles += 1
    ind = banks.index(max(banks))
    blocks = banks[ind]
    banks[ind] = 0
    for _ in range(blocks, 0, -1):
        ind += 1
        if ind == nbanks:
            ind = 0
        banks[ind] += 1
    if banks in configs:
        print "Solution is %d" % cycles
        break
    else:
        configs.append(banks[:])
